Added prayer plugin for kotlin

tlf30 opened the pull request:
Untested. I think it works. Would love if someone could test it. 

~Trevor

tlf30 committed:
Ported prayer plugin to kotlin

garyttierney commented on game/plugin/skills/prayer/src/data.kt:
```diff
@@ -0,0 +1,78 @@
+    SMITE(685, 52, 100, 0.2);
+
+    companion object {
+        fun findByButton(button: Int): Prayer? = Prayer.values().find { prayer -> prayer.button == button }
```
`Enum#values()` returns a copy of the array every time.  Keeping a local copy of `Prayer.values` would be better.

garyttierney commented on game/plugin/skills/prayer/src/data.kt:
```diff
@@ -0,0 +1,78 @@
+    OURG_BONES(4834, 140.0);
+
+    companion object {
+        fun findById(id: Int): Bone? = Bone.values().find { bone -> bone.id == id }
```
Same as Prayer.

garyttierney commented on game/plugin/skills/prayer/src/data.kt:
```diff
@@ -0,0 +1,78 @@
+fun updatePrayer(player: Player, prayer: Prayer?) {
+    //Clear active prayer
+    if (prayer != null) {
+        if (player.getCurrentPrayer() == prayer) {
```
Players can have multiple Prayers active, I think.  An `EnumSet` would be a better way to store active ones.

garyttierney reviewed (dismissed):
Again, mostly LGTM with a few nitpicks.

tlf30 committed:
Fix to allow multiple prayers.
Fixed not caching enum values.

tlf30 commented:
How does that look @garyttierney??

garyttierney commented on game/plugin/skills/prayer/src/data.kt:
```diff
@@ -0,0 +1,75 @@
+val PLAYER_PRAYERS = hashMapOf<Player, ArrayList<Prayer>?>()
+
+fun Player.getCurrentPrayers(): ArrayList<Prayer> {
+    PLAYER_PRAYERS[this] = PLAYER_PRAYERS[this] ?: ArrayList<Prayer>(); //Make sure that the array list exists
```
Any thoughts on a more generic way of handling these scenarios @Major- ?

Major- replied:
Ye we need some sort of replacement for the attributes system, this is fine for now.

Although I think it should be a `Set` not `ArrayList` - we don't need ordering, right?

tlf30 replied:
Good catch, yes it does not need ordering.

garyttierney committed:
Remove now invalid packageName from prayer plugin

garyttierney committed:
Fix code-style in prayer plugin

Adds named arguments to enum declarations along with lookup functions, fixes
continuation indent, and replaces Map<Player, ArrayList<Prayer>> with a
SetMultimap<Player, Prayer>.

garyttierney committed:
Clean up remaining issues on prayer plugin

garyttierney committed:
Use a map for bone and prayer lookups
